### PR TITLE
[MXNET-244] Fixes for cross compilation in ARM

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
See Jira ticket

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by piiswrong** [comment]
@larroy ping

**Comment by larroy** [comment]
Hi, sorry been in holiday and busy with the machine learning conference. We will retake this in the following weeks.

**Comment by szha** [comment]
ping

**Comment by larroy** [comment]
please merge

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -77,7 +82,7 @@ build_armv6() {
         -DUSE_OPENCV=OFF \
         -DUSE_OPENMP=OFF \
         -DUSE_SIGNAL_HANDLER=ON \
-        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
```
Comment (lebeg) [inline ci/docker/runtime_functions.sh]: I think this should stay Release, otherwise it's too big

File: ci/docker/runtime_functions.sh [hunk 2]
```diff
@@ -40,19 +40,24 @@ build_jetson() {
...
+    pushd .
+    python setup.py bdist_wheel --universal
     WHEEL=`readlink -f dist/*.whl`
     TMPDIR=`mktemp -d`
     unzip -d $TMPDIR $WHEEL
     rm $WHEEL
     cd $TMPDIR
     mv *.data/data/mxnet/libmxnet.so mxnet
+    cd $TMPDIR
+    #zip -r $WHEEL $TMPDIR
```
Comment (lebeg) [inline ci/docker/runtime_functions.sh]: Maybe choose one style? I think ```zip -r $WHEEL $TMPDIR``` is good

File: ci/docker/Dockerfile.build.armv7 [hunk 3]
```diff
@@ -20,13 +20,13 @@
...
-ENV ARCH armv71
+ENV ARCH armv7
 ENV CC /usr/bin/arm-linux-gnueabihf-gcc
 ENV CXX /usr/bin/arm-linux-gnueabihf-g++
 
-RUN apt-get update && \
-    apt-get install -y libopenblas-dev:armhf && \
-    rm -rf /var/lib/apt/lists/*
+RUN apt-get update
+RUN apt-get install -y libopenblas-dev:armhf unzip
```
Comment (lebeg) [inline ci/docker/Dockerfile.build.armv7]: Let's build openblas ouself like for other arm platforms?

File: tests/python/unittest/test_ndarray.py [hunk 4]
```diff
@@ -711,9 +711,8 @@ def get_values(ensure_unique):
                  k=dat_size*dat_size*dat_size*dat_size, is_ascend=False)
     assert_almost_equal(nd_ret_argsort, gt)
 
-    # test topk with a big shape
-    a = mx.nd.arange(0, 54686454, step=1, repeat=1)
-    assert_almost_equal(a.topk(k=54686454).asnumpy(), a.asnumpy()[::-1])
+    a = mx.nd.arange(0, 1024, step=1, repeat=1)
```
Comment (lebeg) [inline tests/python/unittest/test_ndarray.py]: This shouldn't be part of the change, right?

Comment (larroy) [reply tests/python/unittest/test_ndarray.py]: it should, so we can pass tests in pi

File: ci/docker/Dockerfile.build.armv7 (hunk 3 above)
Comment (larroy) [reply ci/docker/Dockerfile.build.armv7]: sure, I would make another PR for that.

File: ci/docker/runtime_functions.sh (hunk 2 above)
Comment (larroy) [reply ci/docker/runtime_functions.sh]: I think there were problems, this works.

File: ci/docker/runtime_functions.sh (hunk 1 above)
Comment (larroy) [reply ci/docker/runtime_functions.sh]: done

File: tests/python/unittest/test_ndarray.py (hunk 4 above)
Comment (marcoabreu) [reply tests/python/unittest/test_ndarray.py]: As stated before, this test must not be changed as it's there for a reason. Alternatively, provide a C++ test to cover this case.

Comment (marcoabreu) [reply tests/python/unittest/test_ndarray.py]: See https://github.com/apache/incubator-mxnet/pull/9716 for reference
